go/buildhelper/helpertypes/types.go

```diff
@@ -0,0 +1,130 @@
+package helpertypes
+
+import (
+	"bytes"
+	"compress/gzip"
+	b64 "encoding/base64"
+	"io/ioutil"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/obscuronet/obscuro-playground/go/buildhelper/buildconstants"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+)
+
+var IsRealEth bool
+
+func UnpackL1Tx(tx *types.Transaction) *obscurocommon.L1TxData {
+	if !IsRealEth {
+		t := obscurocommon.TxData(tx)
+		return &t
+	}
+	// ignore transactions that are not calling the contract
+	if tx.To() == nil || tx.To().Hex() != buildconstants.CONTRACT_ADDRESS.Hex() || len(tx.Data()) == 0 {
```

Comment:
@otherview I think `len(tx.Data())` being 0 should be handled the same way as other malformed txs (so a panic for now).

Reply:
I'm not sure.. Would that not break execution if there was a tx that did not call the contract ?
I think the expected behavior is to ignore tx's that we can't unpack, right ?

Reply:
I guess I'm just highlighting the inconsistency. It feels like we're happy to swallow that error, but not ones further down for malformed tx datas.

Reply:
That's a good point raised here - I'm going to bring that up in chat so we're clear on the approach

Reply:
Adding a log so we're on top of transactions that are ignored 👍